v2/cf-nodejs-logging-support/src/lib/logger/record-factory.ts

```diff
@@ -7,16 +7,32 @@ import ResponseAccessor from "../middleware/response-accessor";
 import ReqContext from "./context";
 
 export default class RecordFactory {
+    private static REDUCED_PLACEHOLDER = "redacted";
 
     // init a new record and assign fields with output "msg-log"
     static buildMsgRecord(args: Array<any>, context?: ReqContext): any {
 
-        const msgLogFields = Config.getInstance().getMsgFields();
+        const configInstance = Config.getInstance();
+        const msgLogFields = configInstance.getMsgFields();
         let record: any = {
             "level": "info",
         };
 
         msgLogFields.forEach(field => {
+            if (field.disable) {
+                return;
+            }
+
+            if (!field._meta?.isEnabled == false) {
+                // if enVarRedacted and the value of the field is present set to "redacted", if not ommit
+                if (field._meta?.isRedacted == true) {
+                    let fieldValue = this.getFieldValue(field, record);
+                    if (fieldValue) {
+                        record[field.name] = this.REDUCED_PLACEHOLDER;
+                    }
+                }
+                return;
+            }
             record[field.name] = this.getFieldValue(field, record);
```

Comment:
* If `isEnabled` is `true` and `isRedacted` is `false` the field does not get added to the record due to `return` in line 34. ⚡️
* If `isEnabled` is `false` line 35 gets executed and the field gets added to the record. ⚡️

Please see comment above. I think the logic can be simplified accordingly.